Objects/abstract.c

```diff
@@ -495,6 +500,30 @@ _Py_add_one_to_index_C(int nd, Py_ssize_t *index, const Py_ssize_t *shape)
     }
 }
 
+int
+PyBuffer_SizeFromFormat(const char *format)
+{
+    PyObject *result;
+    int size;
+
+    structmodule = PyImport_ImportModule("struct");
+    if (structmodule == NULL)
+        return NULL;
```

Comment:
wait, NULL is a pointer. The function return type is int: you should return -1 in case of error, no? Same code for others if below.